test/ButtonInputSpec.js

```diff
@@ -72,11 +70,9 @@ describe('ButtonInput', () =>{
     assert.notInstanceOf(result, Error);
   });
 
-  it('does not allow elements for children', function () {
+  it('allows elements as children', function () {
     ReactTestUtils.renderIntoDocument(
       <ButtonInput><span>blah</span></ButtonInput>
     );
-
-    shouldWarn('propType: Invalid');
```

Comment:
Well, it's still a custom prop type validation for children/value. This does accomplish a semi-goal of asserting that it won't throw a spurious false positive here. I'm okay with leaving it in place.